services/shotgrid_common/utils.py

```diff
@@ -184,9 +264,10 @@ def create_sg_entities_in_ay(
     project_entity.folder_types = new_folder_types
 
     # Add Shotgrid Statuses to Project Entity
-    for status in get_sg_statuses(sg_session):
-        status_short_name, status_name = status
-        project_entity.statuses.create(status_name, short_name=status_short_name)
+    ay_status_codes = [s.short_name.lower() for s in list(project_entity.statuses)]
+    for status_code, status_name in get_sg_statuses(sg_session).items():
+        if status_code.lower() not in ay_status_codes:
+            project_entity.statuses.create(status_name, short_name=status_code)
```

Comment:
I like this a lot but somehow feel this should be done in separate PR. I believe that since we are dealing with attributes mapping, statuses and tags synchronization should be separated. I will do it during the merging process as well as it had been done before.

Reply:
not sure I follow, what do you mean it had been done before?

Reply:
I just pushed another big pass of the code to clean up a lot of the variables to be consistent across all the codebase so it's easier to follow